airflow/models.py

```diff
@@ -3455,14 +3366,176 @@ def find(dag_id, run_id=None, state=None, external_trigger=None, session=None):
         qry = session.query(DR).filter(DR.dag_id == dag_id)
         if run_id:
             qry = qry.filter(DR.run_id == run_id)
+        if execution_date:
+            qry = qry.filter(DR.execution_date == execution_date)
         if state:
             qry = qry.filter(DR.state == state)
         if external_trigger:
             qry = qry.filter(DR.external_trigger == external_trigger)
-        dr = qry.all()
+        dr = qry.order_by(DR.execution_date).all()
```

Comment:
Is this `order_by` necessary? I'm sure it adds overhead -- might be better to let users sort the list after calling `find()` if they need it sorted rather than sort it every time.